src/main/java/com/cleanroommc/modularui/widgets/DropDownMenu.java

```diff
@@ -0,0 +1,314 @@
+        menu.setEnabled(false);
+        return Result.SUCCESS;
+    }
+
+    @Override
+    public void draw(GuiContext context) {
+        super.draw(context);
+        Area area = getArea();
+        int smallerSide = area.width > area.height ? area.height : area.width;
+        if (menu.getSelectedItem() != null) {
+            menu.getSelectedItem().setEnabled(true);
+            menu.getSelectedItem().drawBackground(context);
+            menu.getSelectedItem().draw(context);
+            menu.getSelectedItem().drawForeground(context);
+        } else {
+            NONE.draw(context, 0, 0, area.width - smallerSide, area.height);
+        }
+
+        if (menu.isOpen()) {
+            arrowOpened.draw(context, area.width - smallerSide , 0, smallerSide, smallerSide);
+        } else {
+            arrowClosed.draw(context, area.width - smallerSide , 0, smallerSide, smallerSide);
+        }
+    }
+
+    @Override
+    public DropDownMenu background(IDrawable... background) {
+        menu.background(background);
+        return super.background(background);
+    }
+
+    public static enum DropDownDirection {
+        UP(0, -1),
+        DOWN(0, 1);
+
+        private int xOffset = 0;
+        private int yOffset = 0;
+
+        private DropDownDirection(int xOffset, int yOffset) {
+            this.xOffset = xOffset;
+            this.yOffset = yOffset;
+        }
+
+        public int getXOffset() {
+            return xOffset;
+        }
+
+        public int getYOffset() {
+            return yOffset;
+        }
+    }
+
+    private static class DropDownWrapper extends ScrollWidget<DropDownWrapper> {
+        private DropDownDirection direction = DropDownDirection.DOWN;
+        private int maxItemsOnDisplay = 10;
+        private List<IWidget> children = new ArrayList<>();
+        private boolean open;
+        private int count = 0;
+        private int currentIndex = -1;
+
+        public void setDropDownDirection(DropDownDirection direction) {
+            this.direction = direction;
+        }
+
+        @Override
+        public void onFrameUpdate() {
+            if (!open) {
+                setEnabled(false);
+            }
+        }
+
+        @Override
+        public void afterInit() {
+            super.afterInit();
+            rebuild();
+        }
+
+        public void setOpened(boolean open) {
+            this.open = open;
+            rebuild();
+        }
+
+        public boolean isOpen() {
+            return open;
+        }
+
+        public void addChoice(Function<Integer, DropDownItem> itemGetter) {
+            children.add(itemGetter.apply(count));
+            count++;
+        }
+
+        public int getCurrentIndex() {
+            return currentIndex;
+        }
+
+        public void setCurrentIndex(int currentIndex) {
+            this.currentIndex = currentIndex;
+        }
+
+        @Override
+        public @NotNull List<IWidget> getChildren() {
+            return children;
```

Comment:
this probably should be wrapped with `Collections.unmodifiableList` to prevent `count` from going incorrect